messaged.c

```diff
@@ -184,16 +186,47 @@ send_message(struct context *ctx, struct contact *con)
 		buf[size - 1] = '\0';
 		size--;
 	}
-	msg_send(ctx, buf, con->name);
+	/* These characters must be escaped. */
+	if (NULL != strchr(buf, '<') || NULL != strchr(buf, '&')) {
+		/* Get a new string. */
+		escaped = escape_tag(buf);
+	}
+	msg_send(ctx, escaped ? escaped : buf, con->name);
+	free(escaped);
 
-	/* Write message to the out file, that the use see its own messages. */
+	/* Write message to the out file, letting the user see its own messages. */
 	prepare_prompt(prompt, sizeof prompt, ctx->jid);
 	if (write(con->out, prompt, strlen(prompt)) == -1) return false;
 	if (write(con->out, buf, size) == -1) return false;
 	if (write(con->out, "\n", 1) == -1) return false;
 
 	return true;
 }
+char *
+escape_tag(char *string)
+{
+	/* allocate the amount of space that we'll need */
+	/* One byte for the null character */
+	size_t length = 1;
+	size_t i;
```

Comment:
Put the `size_t i` definition in the for-loop, like...

`for (size_t i = 0; ...`